src/imitation/util/serialize.py

```diff
@@ -20,10 +20,20 @@ class Serializable(ABC):
   @abstractmethod
   def load(cls, directory):
     """Load object plus weights from directory."""
+    with open(os.path.join(directory, 'loader'), 'rb') as f:
+      load_cls = pickle.load(f)
+    return load_cls.load(directory)
 
   @abstractmethod
   def save(self, directory):
     """Save object and weights to directory."""
+    os.makedirs(directory, exist_ok=True)
+
+    load_path = os.path.join(directory, 'loader')
+    with open(load_path + '.tmp', 'wb') as f:
+      pickle.dump(type(self), f)
+    # Ensure atomic write
+    os.replace(load_path + '.tmp', load_path)
```

Comment:
This was to support the use case of calling `save` repeatedly with the same directory. `tf.train.Checkpoint` explicitly supports this use case: repeated calls to `save` just add a new checkpoint file. If the code got interrupted in the middle of saving a checkpoint, I don't want it to corrupt previously saved checkpoints in that directory.

Having said that, we might want to just not support this use case, since there's no guarantee that implementations other than `LayersSerializable` would support this.